### PR TITLE
Implemented Token Management Transactions Outcome Parser

### PR DESCRIPTION
**Pull request by popenta**
Implemented the transaction parser for Token Management Transactions. This is meant to be used together with the `TokenManagementTransactionsFactory`.

### CODE REVIEW

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+            () => {
+                parser.parseIssueFungible(txOutcome);
+            },
+            ErrParseTransactionOutcome,
+            "encountered signalError: ticker name is not valid (user error)",
+        );
+    });
+
+    it("should test parse issue fungible", () => {
+        const identifer = "ZZZ-9ee87d";
```
Comment (andreibancioiu) [inline src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts:31]: Typo (and below).

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+    it("should test ensure error", () => {
+        const event = new TransactionEvent({
+            address: "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u",
+            identifier: "signalError",
+            topics: ["Avk0jZ1kR+l9c76wQQoYcu4hvXPz+jxxTdqQeaCrbX8=", "dGlja2VyIG5hbWUgaXMgbm90IHZhbGlk"],
+            data: Buffer.from("QDc1NzM2NTcyMjA2NTcyNzI2Zjcy", "base64"),
+        });
+
+        const logs = new TransactionLogs({ events: [event] });
+        const scResult = new SmartContractResult({});
```
Comment (andreibancioiu) [inline src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts:18]: Does the test work if we don't pass `smartContractResults`? If so, let's simplify it (better for advertising the parsers).

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+            identifier: "issue",
+            topics: [base64Identifier, "U0VDT05E", "Wlpa", "RnVuZ2libGVFU0RU", "Ag=="],
+        });
+
+        const logs = new TransactionLogs({
+            address: "erd18s6a06ktr2v6fgxv4ffhauxvptssnaqlds45qgsrucemlwc8rawq553rt2",
+            events: [event],
+        });
+
+        const scResult = new SmartContractResult({});
```
Comment (andreibancioiu) [inline src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts:45]: Also here (and below), if possible to not pass `scResult` (if not needed), let's do that. On the other hand, it's worth to emphasize that the parser is able to gather logs from the contract results, as well - and you have an example for that below :+1:

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+import { bufferToBigInt } from "../smartcontracts/codec/utils";
+
+export class TokenManagementTransactionsOutcomeParser {
+    constructor() {}
+
+    parseIssueFungible(transactionOutcome: TransactionOutcome): { tokenIdentifier: string } {
+        this.ensureNoError(transactionOutcome.transactionLogs.events);
+
+        const event = this.findSingleEventByIdentifier(transactionOutcome, "issue");
+        const identifer = this.extractTokenIdentifier(event);
```
[the remark this answers is not in the document]
Comment (popenta) [reply src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts:13]: Fixed

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts (hunk 1 above)
Comment (popenta) [reply src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts:31]: Fixed

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts (hunk 2 above)
Comment (popenta) [reply src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts:18]: It does work. Will remove empty objects.

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts (hunk 3 above)
Comment (popenta) [reply src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.spec.ts:45]: Indeed. Changed!

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+
+        const event = this.findSingleEventByIdentifier(transactionOutcome, "ESDTSetRole");
+        const userAddress = event.address;
+        const tokenIdentifier = this.extractTokenIdentifier(event);
+
+        const encodedRoles = event.topics.slice(3);
+
+        let roles: string[] = [];
+        for (const role of encodedRoles) {
+            const decodedRole = Buffer.from(role, "base64");
```
[the remark this answers is not in the document]
Comment (popenta) [reply src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts:86]: Done

File: src/transactionsOutcomeParsers/resources.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
+export class TransactionEvent {
```
[the remark this answers is not in the document]
Comment (popenta) [reply src/transactionsOutcomeParsers/resources.ts:1]: Done!

**Review by andreibancioiu — DISMISSED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Pull request merged**